backend/pkg/database/gorm/scan.go

```diff
@@ -173,38 +189,57 @@ func (s *ScansTableHandler) SaveScan(scan models.Scan) (models.Scan, error) {
 		}
 	}
 
+	if dbScan.Revision != nil {
+		scan.Revision = utils.PointerTo(*dbScan.Revision + 1)
+	} else {
+		scan.Revision = utils.PointerTo(1)
+	}
+
```

Comment:
I think I'll need to create an interface and add GetRevision and SetRevision functions for each object type in the api package if we want to go down that route. Happy to refactor.

Reply:
why it can't be done with a function that gets 2 pointer to int (existing, new -->`dbScan.Revision` and `scan.Revision` in that example) and just set the new one?

Reply:
ah thats a good point, I forgot that because these are all pointers we could do that